src/lib/flow/sol-flow-static.c

```diff
@@ -143,6 +143,8 @@ dispatch_connect_in(struct sol_flow_node *node, uint16_t port, uint16_t conn_id,
 static int
 dispatch_disconnect_out(struct sol_flow_node *node, uint16_t port, uint16_t conn_id, const struct sol_flow_port_type_out *port_type)
 {
+    SOL_FLOW_PORT_TYPE_IN_API_CHECK(port_type, SOL_FLOW_PORT_TYPE_OUT_API_VERSION, 0);
```

Comment:
The port versions (in and out) are checked at connect_nodes(). I removed the checks at  the dispatch_*_*() functions.